src/DataWrangling/JRA55/JRA55_field_time_series.jl

```diff
@@ -69,44 +70,60 @@ function compute_bounding_indices(longitude, latitude, grid, LX, LY, λc, φc)
     return i₁, i₂, j₁, j₂, TX
 end
 
-struct JRA55NetCDFBackend <: AbstractInMemoryBackend{Int}
+struct JRA55NetCDFBackend{M} <: AbstractInMemoryBackend{Int}
     start :: Int
     length :: Int
+    metadata :: M
 end
 
+Adapt.adapt_structure(to, b::JRA55NetCDFBackend) = JRA55NetCDFBackend(b.start, b.length, nothing)
+
 """
     JRA55NetCDFBackend(length)
 
 Represents a JRA55 FieldTimeSeries backed by JRA55 native .nc files.
 """
-JRA55NetCDFBackend(length) = JRA55NetCDFBackend(1, length)
+JRA55NetCDFBackend(length, metadata::Metadata) = JRA55NetCDFBackend(1, length, metadata)
+JRA55NetCDFBackend(start::Integer, length::Integer) = JRA55NetCDFBackend(start, length, nothing)
+
+# Metadata - agnostic constructor
+JRA55NetCDFBackend(length) = JRA55NetCDFBackend(1, length, nothing)
 
 Base.length(backend::JRA55NetCDFBackend) = backend.length
 Base.summary(backend::JRA55NetCDFBackend) = string("JRA55NetCDFBackend(", backend.start, ", ", backend.length, ")")
 
-const JRA55NetCDFFTS = FlavorOfFTS{<:Any, <:Any, <:Any, <:Any, <:JRA55NetCDFBackend}
+const JRA55NetCDFFTS              = FlavorOfFTS{<:Any, <:Any, <:Any, <:Any, <:JRA55NetCDFBackend}
+const JRA55NetCDFFTSRepeatYear    = FlavorOfFTS{<:Any, <:Any, <:Any, <:Any, <:JRA55NetCDFBackend{<:Metadata{<:JRA55RepeatYear}}}
+const JRA55NetCDFFTSMultipleYears = FlavorOfFTS{<:Any, <:Any, <:Any, <:Any, <:JRA55NetCDFBackend{<:Metadata{<:JRA55MultipleYears}}}
 
-# TODO: This will need to change when we add a method for JRA55MultipleYears
-function set!(fts::JRA55NetCDFFTS, path::String=fts.path, name::String=fts.name)
+# Note that each file should have the variables
+#   - ds["time"]:     time coordinate 
+#   - ds["lon"]:      longitude at the location of the variable
+#   - ds["lat"]:      latitude at the location of the variable
+#   - ds["lon_bnds"]: bounding longitudes between which variables are averaged
+#   - ds["lat_bnds"]: bounding latitudes between which variables are averaged
+#   - ds[shortname]:  the variable data
 
-    ds = Dataset(path)
+# Simple case, only one file per variable, no need to deal with multiple files
+function set!(fts::JRA55NetCDFFTSRepeatYear) 
```

Comment:
what does this method do? `set!(fts)` without second argument? What does it set it to?

Reply:
all the information to set the field are contained on the backend and in the time-indices in memory. So basically this command is setting it from stored data

Reply:
It's called here https://github.com/CliMA/Oceananigans.jl/blob/fe5f37b2cfd0ff3ed6e882b1e0496c9546cef991/src/OutputReaders/field_time_series_indexing.jl#L335

Reply:
Gotcha!

Reply:
It might be cleaner or more explicit to have a second argument to this function, for example, a new backend or index range that is being set to.